python/fledge/services/core/api/certificate_store.py

```diff
@@ -79,17 +82,41 @@ async def upload(request):
     key_file = data.get('key')
     cert_file = data.get('cert')
     allow_overwrite = data.get('overwrite', '0')
-
     # accepted values for overwrite are '0 and 1'
+    should_overwrite = False
     if allow_overwrite in ('0', '1'):
         should_overwrite = True if int(allow_overwrite) == 1 else False
     else:
         raise web.HTTPBadRequest(reason="Accepted value for overwrite is 0 or 1")
-
+    
     if not cert_file:
         raise web.HTTPBadRequest(reason="Cert file is missing")
 
     cert_filename = cert_file.filename
+
+    # default installed auth cert keys can be deleted, for matching/debugging disallow overwrite
+    if cert_filename in ['admin.cert', 'admin.key', 'user.cert', 'user.key', 'fledge.key', 'fledge.cert', 'ca.key',
+                         'ca.cert']:
+        if request.is_auth_optional:
+            _logger.warning(FORBIDDEN_MSG)
+            raise web.HTTPForbidden(reason=FORBIDDEN_MSG, body=json.dumps({"message": FORBIDDEN_MSG}))
+        else:
+            if not request.user_is_admin:
+                msg = "admin role permissions required to overwrite the default installed auth/TLS certificates."
+                _logger.warning(msg)
+                raise web.HTTPForbidden(reason=msg, body=json.dumps({"message": msg}))
+    # note.. We are not checking if HTTPS enabled or auth mechanism?
+    # Here, in secured instance, we are simply disallowing non-admin user to overwrite/import configured TLS/CA certificates
+    if request.user and not request.user_is_admin:
+        cf_mgr = ConfigurationManager(connect.get_storage_async())
+        cat = await cf_mgr.get_category_all_items(category_name='rest_api')
+        configured_ca_and_tls_certs = [cat['certificateName']['value'], cat['authCertificateName']['value']]
+        if cert_filename and cert_filename.rpartition('.')[0] in configured_ca_and_tls_certs:  # we better disallow any extension with those names instead of [1]/endswith .cert
+            msg = 'Certificate with name {} is configured to be used, ' \
+                  'An `admin` role permissions required to add/overwrite.'.format(cert_filename)
+            _logger.warning(msg)
+            raise web.HTTPForbidden(reason=msg, body=json.dumps({"message": msg}))
+
     key_valid_extensions = ('.key', '.pem')
     cert_valid_extensions = ('.cert', '.cer', '.crt', '.json', '.pem')
```

Comment:
@ashish-jabble Please file a jira to  allow .p12, .pfx and .der extensions. May be .csr/.crl too.

Reply:
FOGL-5480 JIRA raised.